tests/test_cnes_sonar_scanner.py

```diff
@@ -0,0 +1,517 @@
+        :param language_key: language key for SonarQube
+        :param folder: folder name, relative to the tests/ folder
+        :param sensors_info: array of lines of sensors to look for in the scanner output
+        :param project_key: project key (sonar.project_key of sonar-project.properties)
+        :param nb_issues: number of issues with the Sonar way Quality Profile
+        :param cnes_qp: (optional) name of the CNES Quality Profile to apply, if any
+        :param nb_issues_cnes_qp: (optional) number of issues with the CNES Quality Profile, if specified
+
+        Example (not a doctest):
+            sensors = (
+                "INFO: Sensor CheckstyleSensor [checkstyle]",
+                "INFO: Sensor FindBugs Sensor [findbugs]",
+                "INFO: Sensor PmdSensor [pmd]",
+                "INFO: Sensor CoberturaSensor [cobertura]"
+            )
+            self.language("Java", "java", "java", sensors, "java-dummy-project", 3, "CNES_JAVA_A", 6)
+        """
+        docker_client = docker.from_env()
+        print(f"Analysing project {project_key}...")
+        output = docker_client.containers.run(cls._SONAR_SCANNER_IMAGE, f"-Dsonar.projectBaseDir=/usr/src/tests/{folder}",
+            auto_remove=True,
+            environment={"SONAR_HOST_URL": cls.SONARQUBE_URL},
+            network=cls.SONARQUBE_NETWORK,
+            user=f"{os.getuid()}:{os.getgid()}",
+            volumes={
+                f"{cls._PROJECT_ROOT_DIR}": {'bind': '/usr/src', 'mode': 'rw'},
+                f"{cls._PROJECT_ROOT_DIR}/.sonarcache": {'bind': '/opt/sonar-scanner/.sonar/cache', 'mode': 'rw'}
+            }).decode("utf-8")
+        print(output)
+        # Make sure all non-default for this language plugins were executed by the scanner
+        for sensor_line in sensors_info:
+            # Hint: if this test fails, a plugin may not be installed correctly or a sensor is not triggered when needed
+            assert sensor_line in output
+        # Wait for SonarQube to process the results
+        time.sleep(8)
+        # Check that the project was added to the server
+        output = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/projects/search?projects={project_key}",
+                        auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()
+        # Hint: if this test fails, the project is not on the server
+        assert output['components'][0]['key'] == project_key
+        # Get the number of issues of the project
+        output = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/issues/search?componentKeys={project_key}",
+                        auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()['issues']
+        issues = [ issue for issue in output if issue['status'] in ('OPEN', 'TO_REVIEW') ]
+        # Hint: if this test fails, there should be {nb_issues issues} on the {language_name} dummy project with the Sonar way QP but {len(issues)} were found
+        assert len(issues) == nb_issues
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6252273)